src/coord/src/coord.rs

```diff
@@ -5492,8 +5492,12 @@ mod timeline {
         /// `self.read_ts()`.
         pub fn write_ts(&mut self) -> T {
             match &self.state {
-                TimestampOracleState::Writing(ts) => ts.clone(),
```

Comment:
I think we want the TimestampOracle to not change. It's pretty carefully considered, and changing anything in it requires reconsidering its guarantees and edge cases. I recommend implementing this PR with the existing oracle API, which I think would be to always call `read_ts()` immediately after calling `write_ts()`